core/src/main/scala/sttp/openai/requests/completions/chat/ToolCall.scala

```diff
@@ -0,0 +1,16 @@
+package sttp.openai.requests.completions.chat
+
+import sttp.openai.json.SnakePickle
+
+/** @param id
+  *   The ID of the tool call.
+  * @param type
+  *   The type of the tool. Currently, only function is supported.
+  * @param function
+  *   The function that the model called.
+  */
+case class ToolCall(id: String, `type`: String, function: FunctionCall)
```

Comment:
If in future we plan to implement also tools other than "function" - maybe `ToolCall` should be a trait, and we should have `FunctionToolCall` instead of this case class, implementing `ToolCall` trait?

Reply:
+1, and use the `type` field as a discriminator.